lib/harness/ssh.js

```diff
@@ -89,14 +82,18 @@ class SSHConnector {
                                     process.stdout.write( data );
                                 }
                                 buffer += data;
+                                if ( options.setup && data.includes( options.setup.wait_for ) ) {
+                                    c.end();
+                                    resolve( buffer );
+                                }
                             })
                             .stderr.on('data', function (data) {
                                 if( verbose )
                                 {
                                     process.stderr.write( data );
                                 }
                                 // We add stderr to same output.
-                                buffer += data;
+                                resolve(data);
```

Comment:
This might not always return the desired output from `_JSSSHExec()` (if it's not called from `setup()`). I'm not 100% sure but I think in some cases, this could break other functionality. Could you put this in `.on('close'` instead?

Reply:
Yeah sure!